R/redirectSource.R

```diff
@@ -0,0 +1,143 @@
+#' redirect
+#'
+#' Redirect a given dataset type to a different source folder.
+#' The redirection is local, so it will be reset when the current function call returns. See
+#' example for more details.
+#'
+#' @param type Dataset name, e.g. "Tau" for \code{\link{readTau}}
+#' @param target Either path to the new source folder that should be used instead of the default,
+#' or NULL to remove the redirection, or a vector of paths to files which are then symlinked
+#' into a temporary folder that is then used as target folder; if the vector is named the names
+#' are used as relative paths in the temporary folder, e.g. target = c(`a/b/c.txt` = "~/d/e/f.txt")
+#' would create a temporary folder with subfolders a/b and there symlink c.txt to ~/d/e/f.txt.
+#' @param local The scope of the redirection, passed on to setConfig. Defaults to the current function.
+#' Set to an environment for more control or to FALSE for a permanent/global redirection.
+#' @param linkOthers If target is a list of files, whether to symlink all other files in the original
+#' source folder to the temporary folder.
+#' @return Invisibly, the source folder that is now used for the given type
+#' @author Pascal Sauer
+#' @examples \dontrun{
+#' f <- function() {
+#'   redirect("Tau", target = "~/TauExperiment")
+#'   # the following call will change directory
+#'   # into ~/TauExperiment instead of <getConfig("sourcefolder")>/Tau
+#'   readSource("Tau")
+#' }
+#' f()
+#' # Tau is only redirected in the local environment of f,
+#' # so it will use the usual source folder here
+#' readSource("Tau")
+#' }
+#' @export
```

Comment:
I find that confusing to have two rather similar functions exported (redirect and redirectSource). Lets have a discussion whether we can unify them somehow.